pkgs/cronet_http/lib/src/cronet_client.dart

```diff
@@ -267,10 +267,19 @@ class CronetClient extends BaseClient {
         .map((key, value) => MapEntry(key.toLowerCase(), value.join(',')));
 
     final contentLengthHeader = responseHeaders['content-length'];
+    int? contentLength;
+    if (contentLengthHeader != null) {
+      if (!RegExp(r'^\d+$').hasMatch(contentLengthHeader)) {
+        throw ClientException(
+          'Invalid content-length header [$contentLengthHeader].',
+          request.url,
+        );
+      }
+      contentLength = int.parse(contentLengthHeader);
```

Comment:
I was first going to suggest pulling out the RegExp instead of reconstructing it for each request - but I think we can skip the regex entirely.

```suggestion
      contentLength = int.tryParse(contentLengthHeader);
      if (contentLength == null) {
        throw ClientException(
          'Invalid content-length header [$contentLengthHeader].',
          request.url,
        );
      }
```

Reply:
We also need to check that it is non-negative. I used the regex because it matches the production in RFC-2616:
https://datatracker.ietf.org/doc/html/rfc2616#section-14.13

I don't have strong feelings about this though.